### PR TITLE
Allow logs, tidy and tag should use queryUser

### PR DESCRIPTION
**Pull request by danny-may**
Logs, tidy and tag dont rely on a user being on the guild, so they should use queryUser.

### CODE REVIEW

**Comment by danny-may** [comment]
Please excuse the additional files that got included. Github actions are using a newer version of node than they used to which doesnt work with the current version of yarn berry. I had to update yarn and typescript, which came with a few compile errors.

File: src/cluster/command/CommandContext.ts [hunk 1]
```diff
@@ -103,9 +103,20 @@ export class CommandContext<TChannel extends KnownTextableChannel = KnownTextabl
...
     }
+    public async queryUser(options: SlimEntityFindQueryOptions): Promise<ChoiceQueryResult<User>>;
+    public async queryUser(this: GuildCommandContext, options: Omit<SlimEntityFindQueryOptions, 'guild'>): Promise<ChoiceQueryResult<User>>;
+    public async queryUser(options: SlimEntityPickQueryOptions<User>): Promise<ChoiceQueryResult<User>>;
+    public async queryUser(options: SlimEntityQueryOptions<User> | Omit<SlimEntityFindQueryOptions, 'guild'>): Promise<ChoiceQueryResult<User>> {
+        if ('choices' in options)
+            return await this.util.queryUser({ ...options, context: this.message, actors: this.author });
+
+        if ('guild' in options)
+            return await this.util.queryUser({ ...options, context: this.message, actors: this.author });
```
Comment (HunteRoi) [inline src/cluster/command/CommandContext.ts:114]: could be merged into a single condition, right?

File: src/cluster/command/CommandContext.ts [hunk 2]
```diff
@@ -103,9 +103,20 @@ export class CommandContext<TChannel extends KnownTextableChannel = KnownTextabl
 
         throw new Error('Cannot queryMember without a guild!');
     }
+    public async queryUser(options: SlimEntityFindQueryOptions): Promise<ChoiceQueryResult<User>>;
+    public async queryUser(this: GuildCommandContext, options: Omit<SlimEntityFindQueryOptions, 'guild'>): Promise<ChoiceQueryResult<User>>;
```
Comment (HunteRoi) [inline src/cluster/command/CommandContext.ts:107]: what's the point in overwriting the this here?

**Review by HunteRoi — CHANGES_REQUESTED** [review]

File: src/cluster/command/CommandContext.ts (hunk 1 above)
Comment (danny-may) [reply src/cluster/command/CommandContext.ts:114]: sadly not. Logically yes, but typescript complains if you do
image

File: src/cluster/command/CommandContext.ts (hunk 2 above)
Comment (danny-may) [reply src/cluster/command/CommandContext.ts:107]: Its to add the overloads that are accepted. The second overload is for the case where there is a guild associated with the context, so you dont need to supply one.

**Review by HunteRoi — APPROVED** [review]

**Pull request merged**